### PR TITLE
tests and fixes for AWS backend cleaning

### PR DESCRIPTION
**Pull request by ephur**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -16,17 +16,85 @@
...
+def s3_client(aws_credentials):
+    with mock_s3():
+        boto3.setup_default_session()
+        yield boto3.client("s3", region_name="us-west-2")
+
+
+@pytest.fixture(scope="class")
+def dynamodb_client(aws_credentials):
+    with mock_dynamodb2():
+        # boto3.setup_default_session()
```
Comment (ghost) [inline tests/conftest.py]: Why isn't this needed here?

File: tfworker/backends/s3.py [hunk 2]
```diff
@@ -116,3 +116,115 @@ def data_hcl(self, remotes: list) -> str:
...
+                click.secho(f"error deleting state: {e}")
+                raise SystemExit(1)
+            self._clean_locking_state(deployment)
+
+    def delete_with_versions(self, key):
+        """
+        delete_with_versions should handle object deletions, and all references / versions of the object
+
+        note: in initial testing this isn't required, but is inconsistent with how S3 delete markers, and the boto
+        delete object call work there may be some selfurations that require extra handling.
```
Comment (ghost) [inline tfworker/backends/s3.py]: Is `selfurations` supposed to be `configurations`?

File: tests/conftest.py (hunk 1 above)
Comment (ephur) [reply tests/conftest.py]: before using the mocked aws authenticator I was trying to troubleshoot, good catch this is actually no longer needed anywhere.

File: tfworker/backends/s3.py (hunk 2 above)
Comment (ephur) [reply tfworker/backends/s3.py]: probably, updated!

**Review by ghost — APPROVED** [review]

**Pull request merged**